wireapp/wire-ios-mono
# feat: extract developer flags defaults to configuration file - WPB-3039

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3039" title="WPB-3039" target="_blank"><img alt="Task" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10818?size=medium" />WPB-3039</a>  Extract MLS and CoreCrypto Developer flags to Xcconfig file
  </td></table>
  <br />
 












----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Default values for developer flag are hard coded, which means that if a build needs different flag value we need to diverge the source code.

### Solutions

Extract default values to configuration file.

### Dependencies (Optional)

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-build-configuration/pull/67
- [x] https://github.com/wireapp/wire-ios-build-assets/pull/91

### Testing

#### How to Test

Manually:
1. Change value in Developer-Flags.xcconfig under Configuration
2. Run the app and check the value under DeveloperFlags
3. If no value already set, it should read the default one

### Notes (Optional)

As DeveloperFlags are needed through multiple frameworks, we set xcconfig in Utilities so it's accessible.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
